contract/src/airdrop/prepareAirdropKit.js

```diff
@@ -0,0 +1,341 @@
+        addEligibleUsers: M.call(M.arrayOf(M.string())).returns(M.string()),
+      }),
+      public: M.interface('Public Facet', {
+        getTreeRoot: M.call().returns(M.string()),
+
+        claimInclusion: M.call(M.string(), M.arrayOf(M.string())).returns(
+          M.any(),
+        ),
+        // getAirdropIssuer: M.call().returns(IssuerShape),
+        getAirdropTokenDetails: M.call().returns(AirdropIssuerDetailsShape),
+        claim: M.call(M.string()).returns(M.any()),
+      }),
+      helpers: M.interface('Helper Facet', {
+        getClaimPeriodP: M.call().returns(M.promise()),
+
+        getInternalDepositFacet: M.call().returns(DepositFacetShape),
+        createPurse: M.call().returns(PurseShape),
+        setAirdropWaker: M.call().returns(M.promise()),
+      }),
+    },
+    (claimWindow, hash, claimeeStore, internalPurse) => {
+      return {
+        claimWindowTimeframe: claimWindow,
+        rootHash: hash,
+        claimedUsersStore: claimeeStore,
+        internalPurse,
+        totalTokensClaimed: 0n,
+        distributionSchedule,
+        currentEpoch: distributionSchedule[0],
+      };
+    },
+    {
+      helpers: {
+        createPurse() {
+          return this.state.issuer.makeEmptyPurse();
+        },
+        getInternalDepositFacet() {
+          return this.state.internalPurse.getDepositFacet();
+        },
+        async setAirdropWaker() {
+          await E(contractTimer).setWakeup(
+            privateArgs.claimWindowStartTime,
+            makeWaker('airdropExpirationWaker', () => {
+              // burn tokens from purse
+            }),
+          );
+        },
+        async getClaimPeriodP(p) {
+          return Far('claimPeriodPromise', {
+            getClaimPeriodP: () => p,
+          });
+        },
+      },
+      creator: {
+        makeOpenClaimingWindow() {
+          console.group('---------- inside makeOpenClaimingWindow----------');
+          console.log('------------------------');
+          console.log('this.state::', this.state);
+          console.log('------------------------');
+          console.log('::');
+          console.log('------------------------');
+          console.groupEnd();
+
+          /** @type {OfferHandler} */
+          const startAirdropHandler = async (seat, offerArgs) => {
+            const {
+              give: { Deposit },
+            } = seat.getProposal();
+            atomicRearrange(zcf, harden([[seat, contractSeat, { Deposit }]]));
+
+            console.log(
+              'inside startAirdropHandler',
+              contractSeat.getCurrentAllocation(),
+            );
+
+            seat.exit();
+            void this.facets.helpers.setAirdropWaker();
+            return 'successfully opened claiming window';
+          };
+
+          return zcf.makeInvitation(startAirdropHandler, 'startAirdropHandler');
+        },
+        getPurse() {
+          return airdropPurse;
+        },
```

Comment:
`getPurse` seems like asking for trouble. purses are normally closely held. what is `getPurse` used for, I wonder?

Reply:
`prepare.js` is the successor of ﻿`prepareAirdropKit.js`. You managed to access it before I nuked the latter a moment ago.

that said, I have no intention of exposing the purse in this manner, or in any other manner.